File: Editorial/0058_Length_of_Last_Word/CSharp/1_String_Index_Manipulation.cs
public class Solution {
    public int LengthOfLastWord(string s) {
        // trim the trailing spaces
        int p = s.Length - 1;
        while (p >= 0 && s[p] == ' ') {
            p--;
        }

        // compute the length of last word
        int length = 0;
        while (p >= 0 && s[p] != ' ') {
            p--;
            length++;
        }

        return length;
    }
}

File: Editorial/0000-0100/0058_Length_of_Last_Word/Go/1_String_Index_Manipulation.go
func lengthOfLastWord(s string) int {
    // trim the trailing spaces
    p := len(s) - 1
    for p >= 0 && s[p] == ' ' {
        p--
    }
    // compute the length of last word
    length := 0
    for p >= 0 && s[p] != ' ' {
        p--
        length++
    }
    return length
}
